src/connectors/crowdstrike_devices.py

```diff
@@ -0,0 +1,279 @@
+"""Crowdstrike Devices
+Collect Crowdstrike Device information using a Client ID and Secret
+"""
+
+from runners.helpers import db, log
+from runners.helpers.dbconfig import ROLE as SA_ROLE
+
+from datetime import datetime
+
+import requests
+from .utils import yaml_dump
+
+PAGE_SIZE = 1000
+
+CROWDSTRIKE_AUTH_TOKEN_URL = 'https://api.crowdstrike.com/oauth2/token'
+CROWDSTRIKE_DEVICES_BY_ID_URL = 'https://api.crowdstrike.com/devices/queries/devices-scroll/v1'
+CROWDSTRIKE_DEVICE_DETAILS_URL = 'https://api.crowdstrike.com/devices/entities/devices/v1'
+
+CONNECTION_OPTIONS = [
+    {
+        'name': 'client_id',
+        'title': "Crowdstrike API Client ID",
+        'prompt': "Your Crowdstrike Client ID",
+        'type': 'str',
+        'required': True,
+        'secret': True,
+    },
+    {
+        'name': 'client_secret',
+        'title': "Crowdstrike API Client Secret",
+        'prompt': "Your Crowdstrike API Client Secret",
+        'type': 'str',
+        'secret': True,
+        'required': True,
+    },
+]
+
+LANDING_TABLE_COLUMNS = [
+    ('INSERT_ID', 'NUMBER IDENTITY START 1 INCREMENT 1'),
+    ('SNAPSHOT_AT', 'TIMESTAMP_LTZ(9)'),
+    ('RAW', 'VARIANT'),
+    ('DEVICE_ID', 'STRING'),
+    ('FIRST_SEEN', 'TIMESTAMP_LTZ(9)'),
+    ('SYSTEM_MANUFACTURER', 'STRING'),
+    ('CONFIG_ID_BASE', 'NUMBER'),
+    ('LAST_SEEN', 'TIMESTAMP_LTZ(9)'),
+    ('POLICIES', 'VARIANT'),
+    ('SLOW_CHANGING_MODIFIED_TIMESTAMP', 'TIMESTAMP_LTZ(9)'),
+    ('MINOR_VERSION', 'NUMBER'),
+    ('SYSTEM_PRODUCT_NAME', 'STRING'),
+    ('HOSTNAME', 'STRING'),
+    ('MAC_ADDRESS', 'STRING'),
+    ('PRODUCT_TYPE_DESC', 'STRING'),
+    ('PLATFORM_NAME', 'STRING'),
+    ('EXTERNAL_IP', 'STRING'),
+    ('AGENT_LOAD_FLAGS', 'NUMBER'),
+    ('GROUP_HASH', 'STRING'),
+    ('PROVISION_STATUS', 'STRING'),
+    ('OS_VERSION', 'STRING'),
+    ('GROUPS', 'VARIANT'),
+    ('BIOS_VERSION', 'STRING'),
+    ('MODIFIED_TIMESTAMP', 'TIMESTAMP_LTZ(9)'),
+    ('LOCAL_IP', 'STRING'),
+    ('AGENT_VERSION', 'STRING'),
+    ('MAJOR_VERSION', 'NUMBER'),
+    ('META', 'VARIANT'),
+    ('AGENT_LOCAL_TIME', 'TIMESTAMP_LTZ(9)'),
+    ('BIOS_MANUFACTURER', 'STRING'),
+    ('PLATFORM_ID', 'NUMBER'),
+    ('DEVICE_POLICIES', 'VARIANT'),
+    ('CONFIG_ID_BUILD', 'NUMBER'),
+    ('CONFIG_ID_PLATFORM', 'NUMBER'),
+    ('CID', 'STRING'),
+    ('STATUS', 'STRING'),
+    ('SERVICE_PACK_MINOR', 'NUMBER'),
+    ('PRODUCT_TYPE', 'NUMBER'),
+    ('SERVICE_PACK_MAJOR', 'NUMBER'),
+    ('BUILD_NUMBER', 'NUMBER'),
+    ('POINTER_SIZE', 'NUMBER'),
+    ('SITE_NAME', 'STRING'),
+    ('MACHINE_DOMAIN', 'STRING'),
+    ('OU', 'VARIANT'),
+]
+
+
+def get_col_transform(idx: int) -> str:
```

Comment:
This section is great and I'd love to see it integrated into db.py somehow instead of just being a part of the connector!